### PR TITLE
Cherrypick kfdef

### PR DESCRIPTION
**Pull request by gabrielwen**
Related: kubeflow/kubeflow#3703

Cherrypick #553 onto v0.7-branch.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/565)

### CODE REVIEW

File: kfdef/kfctl_anthos.0.7.0.yaml [hunk 1]
```diff
@@ -277,9 +270,9 @@ spec:
...
     name: seldon-core-operator
-  enableApplications: true
-  packageManager: kustomize
-  skipInitProject: true
-  useBasicAuth: false
-  useIstio: true
-  version: v0.7.0
+  repos:
+  - name: manifests
+    uri: https://github.com/kubeflow/manifests/archive/master.tar.gz
```
Comment (jlewi) [inline kfdef/kfctl_anthos.0.7.0.yaml]: Update to v0.7-branc

File: kfdef/kfctl_aws.0.7.0.yaml [hunk 2]
```diff
@@ -237,10 +251,51 @@ spec:
...
+            name: password
+          username: admin
+      region: us-west-2
+      roles:
+      - eksctl-kubeflow-aws-nodegroup-ng-a2-NodeInstanceRole-xxxxxxx
   repos:
   - name: manifests
-    root: manifests-0.7-branch
-    uri: https://github.com/kubeflow/manifests/archive/v0.7-branch/head.tar.gz
+    uri: https://github.com/kubeflow/manifests/archive/master.tar.gz
```
Comment (jlewi) [inline kfdef/kfctl_aws.0.7.0.yaml]: v0.7-branch

File: kfdef/kfctl_aws_cognito.0.7.0.yaml [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+          certArn: arn:aws:acm:us-west-2:xxxxx:certificate/xxxxxxxxxxxxx-xxxx
+          cognitoAppClientId: xxxxxbxxxxxx
+          cognitoUserPoolArn: arn:aws:cognito-idp:us-west-2:xxxxx:userpool/us-west-2_xxxxxx
+          cognitoUserPoolDomain: your-user-pool
+      region: us-west-2
+      roles:
+      - eksctl-kubeflow-aws-nodegroup-ng-a2-NodeInstanceRole-xxxxx
+  repos:
+  - name: manifests
+    uri: https://github.com/kubeflow/manifests/archive/master.tar.gz
```
Comment (jlewi) [inline kfdef/kfctl_aws_cognito.0.7.0.yaml]: v0.7 branch

File: kfdef/kfctl_existing_arrikto.0.7.0.yaml [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+  - kustomizeConfig:
+      overlays:
+      - application
+      repoRef:
+        name: manifests
+        path: seldon/seldon-core-operator
+    name: seldon-core-operator
+  repos:
+  - name: manifests
+    uri: https://github.com/kubeflow/manifests/archive/master.tar.gz
```
Comment (jlewi) [inline kfdef/kfctl_existing_arrikto.0.7.0.yaml]: v0.7-branch

File: kfdef/kfctl_anthos.0.7.0.yaml (hunk 1 above)
Comment (gabrielwen) [reply kfdef/kfctl_anthos.0.7.0.yaml]: done.

File: kfdef/kfctl_aws.0.7.0.yaml (hunk 2 above)
Comment (gabrielwen) [reply kfdef/kfctl_aws.0.7.0.yaml]: done.

File: kfdef/kfctl_existing_arrikto.0.7.0.yaml (hunk 4 above)
Comment (gabrielwen) [reply kfdef/kfctl_existing_arrikto.0.7.0.yaml]: done.

**Comment by Jeffwan** [comment]
@gabrielwen Hi, just notice the filename was changed? Is this a regular pattern?  Should we same file name? This requires documentation side make a corresponding change.

We have this name in the doc
https://raw.githubusercontent.com/kubeflow/manifests/v0.7-branch/kfdef/kfctl_aws.yaml

Right now, we need to change to 
https://raw.githubusercontent.com/kubeflow/manifests/v0.7-branch/kfdef/kfctl_aws0.7.0.yaml
?

**Comment by gabrielwen** [comment]
> @gabrielwen Hi, just notice the filename was changed? Is this a regular pattern? Should we same file name? This requires documentation side make a corresponding change.
> 
> We have this name in the doc
> https://raw.githubusercontent.com/kubeflow/manifests/v0.7-branch/kfdef/kfctl_aws.yaml
> 
> Right now, we need to change to
> https://raw.githubusercontent.com/kubeflow/manifests/v0.7-branch/kfdef/kfctl_aws0.7.0.yaml
> ?
> 
> I think it's unnecessary to change file name?

that is used to pin to a version.  for example in our doc we use:
https://raw.githubusercontent.com/kubeflow/manifests/fccc0ac/kfdef/kfctl_gcp_iap.0.7.0.yaml

@jlewi could provide you some more insights.

**Pull request merged**